src/scheduler/schedulerServices.tsx

```diff
@@ -0,0 +1,707 @@
+/**
+ * @license
+ * Copyright 2023 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { toast } from 'react-toastify';
+import { requestAPI } from '../handler/handler';
+import { DataprocLoggingService, LOG_LEVEL } from '../utils/loggingService';
+import { toastifyCustomStyle } from '../utils/utils';
+import { JupyterLab } from '@jupyterlab/application';
+import { scheduleMode } from '../utils/const';
+
+interface IPayload {
+  input_filename: string;
+  composer_environment_name: string;
+  output_formats: string[];
+  parameters: string[];
+  cluster_name?: string;
+  serverless_name?: {} | undefined;
+  mode_selected: string;
+  retry_count: number | undefined;
+  retry_delay: number | undefined;
+  email_failure: boolean;
+  email_delay: boolean;
+  email: string[];
+  name: string;
+  schedule_value: string;
+  stop_cluster: boolean;
+  time_zone?: string;
+  dag_id: string;
+}
+
+interface IUpdateSchedulerAPIResponse {
+  status: number;
+  error: string;
+}
+
+interface ISchedulerDagData {
+  dag_id: string;
+  timetable_description: string;
+  is_paused: string;
+  schedule_interval: null | {
+    value: string;
+  };
+}
+interface IDagList {
+  jobid: string;
+  notebookname: string;
+  schedule: string;
+  scheduleInterval: string;
+}
+
+export class SchedulerService {
+  static listClustersAPIService = async (
+    setClusterList: (value: string[]) => void,
+    nextPageToken?: string,
+    previousClustersList?: object
+  ) => {
+    const pageToken = nextPageToken ?? '';
+    try {
+      const serviceURL = `clusterList?pageSize=50&pageToken=${pageToken}`;
+
+      const formattedResponse: any = await requestAPI(serviceURL);
+      let transformClusterListData = [];
+      if (formattedResponse && formattedResponse.clusters) {
+        transformClusterListData = formattedResponse.clusters.map(
+          (data: any) => {
+            return {
+              clusterName: data.clusterName
+            };
+          }
+        );
+      }
+      const existingClusterData = previousClustersList ?? [];
+      //setStateAction never type issue
+      const allClustersData: any = [
+        ...(existingClusterData as []),
+        ...transformClusterListData
+      ];
+
+      if (formattedResponse.nextPageToken) {
+        this.listClustersAPIService(
```

Comment:
This is going to be really slow if we are requesting 50 items at a time trying to exchaustively list all the clusters.  Can we increase the page size and maybe set a item limit?

Reply:
Resolved - Increased page size